dtfs-central-api/src/repositories/payment-repo/payment.repo.ts

```diff
@@ -43,4 +44,34 @@ export const PaymentRepo = SqlDbDataSource.getRepository(PaymentEntity).extend({
       },
     });
   },
+
+  /**
+   * Finds payment entities that are part of the same group as the payment
+   * with the supplied id.
+   * @param paymentId - The payment id
+   * @param reportId - The report id
+   * @returns An array of payment entities in the same group as the specified
+   * payment, including their fee records and associated reports.
+   */
+  async findPaymentsInGroupContainingPaymentByIdAndReportIdWithFeeRecords(paymentId: number, reportId: number): Promise<PaymentEntity[]> {
```

Comment:
Is there a simpler name we could use for this? It's very long and from reading it, it's not clear what it's doing, e.g it looks like:

- It might be getting "payments in a group that contains a payment *and* the group contains a report ID"
- Or it might be getting "payments in a group that contains a payment and the payment has a report ID"

Maybe it could be something like `findPaymentsByIds`?

Will there be a scenario where we want to get payments _without_ fee records?

Reply:
The two scenarios will always return the same thing as grouped payments will always be for the same report.

There are scenarios where we want to fetch payments without fee records, we have the naming convention on our SQL repos that when we include relations we include this as the 'with' part of the name. 

Given this, do you think `findPaymentsInGroupByPaymentIdAndReportIdWithFeeRecords` would be a better name that still fits within these naming conventions?

Reply:
Not sure what to suggest especially if this is the current convention. Long names like this are very hard to read/follow IMO. Appreciate this is not the scope of this PR, maybe object structured params could be nicer e.g

```ts
findPaymentsById({
  paymentId,
  withFeeRecords = false,
  reportId,
});
```

Just an idea off the top of my head, i'm not familiar with the product/service or usage of such functions.

Reply:
I've changed it to the shorter name above but kept the 'with' part of the method name as the relations we include with this particular repo method are a bit more complex than the other repo methods